pkg/test/collector_producer_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package test
+
+import (
+	"net"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	saramamock "github.com/Shopify/sarama/mocks"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/go-ipfix/pkg/collector"
+	"github.com/vmware/go-ipfix/pkg/producer"
+	"github.com/vmware/go-ipfix/pkg/registry"
+)
+
+var (
+	msg1InBytes = []byte{0x3d, 0x8, 0xf1, 0xa0, 0xe9, 0xfc, 0x5, 0x18, 0x1, 0x32,
+		0x7, 0x31, 0x2e, 0x32, 0x2e, 0x33, 0x2e, 0x34, 0x3a, 0x7, 0x35, 0x2e, 0x36,
+		0x2e, 0x37, 0x2e, 0x38, 0x40, 0xd2, 0x9, 0x48, 0xae, 0x2c, 0x50, 0x6, 0x9a,
+		0x1, 0x4, 0x70, 0x6f, 0x64, 0x31, 0xca, 0x1, 0x5, 0x3f, 0x31, 0x32, 0x38,
+		0x33, 0x8a, 0x2, 0x9, 0x31, 0x32, 0x37, 0x2e, 0x30, 0x2e, 0x30, 0x2e, 0x31}
+	msg2InBytes = []byte{0x3d, 0x8, 0xf2, 0xa4, 0xe9, 0xfc, 0x5, 0x18, 0x1, 0x32,
+		0x7, 0x31, 0x2e, 0x32, 0x2e, 0x33, 0x2e, 0x34, 0x3a, 0x7, 0x35, 0x2e, 0x36,
+		0x2e, 0x37, 0x2e, 0x38, 0x40, 0xd2, 0x9, 0x48, 0xae, 0x2c, 0x50, 0x6, 0xb2,
+		0x1, 0x4, 0x70, 0x6f, 0x64, 0x32, 0xca, 0x1, 0x5, 0x3f, 0x30, 0x30, 0x30,
+		0x30, 0x8a, 0x2, 0x9, 0x31, 0x32, 0x37, 0x2e, 0x30, 0x2e, 0x30, 0x2e, 0x31}
+)
+
+func TestCollectorToProducer(t *testing.T) {
+	registry.LoadRegistry()
+	address, err := net.ResolveUDPAddr("udp", "0.0.0.0:4739")
+	if err != nil {
+		t.Error(err)
+	}
+	// Initialize collecting process
+	cp, _ := collector.InitCollectingProcess(address, 1024, 0)
+	// Create a mock Kafka producer
+	kafkaConfig := sarama.NewConfig()
+	kafkaConfig.Version = producer.KafkaConfigVersion
+	kafkaConfig.Producer.Return.Successes = true
+	kafkaConfig.Producer.Return.Errors = true
+
+	mockProducer := saramamock.NewAsyncProducer(t, kafkaConfig)
+	kafkaProducer := producer.NewKafkaProducer(mockProducer, "test-flow-msgs")
+
+	go func() {
+		time.Sleep(time.Second)
```

Comment:
We can use similar to latest code without sleep for this.

Reply:
done